python/activator/middleware_interface.py

```diff
@@ -506,52 +519,94 @@ def get_key(ref):
         for k, g in itertools.groupby(ordered, key=get_key):
             yield k, len(list(g))
 
-    def _get_raw_run_name(self, visit):
-        """Define a run collection specific to a particular visit.
+    def _get_init_output_run(self,
+                             visit: Visit,
+                             date: datetime.date = datetime.datetime.now(datetime.timezone.utc)) -> str:
+        """Generate a deterministic init-output collection name that avoids
+        configuration conflicts.
+
+        Parameters
+        ----------
+        visit : Visit
+            Group of snaps whose processing goes into the run.
+        date : `datetime.date`
+            Date of the processing run (not observation!)
+
+        Returns
+        -------
+        run : `str`
+            The run in which to place pipeline init-outputs.
+        """
+        # Current executor requires that init-outputs be in the same run as
+        # outputs. This can be changed once DM-36162 is done.
+        return self._get_output_run(visit, date)
```

Comment:
No. I avoided adding a line for it because it depends on how we handle init-outputs in the future. Jim Bosch proposed that we create them in the central repo and then import them like we do calibs and templates; if we do that, then the correct behavior depends on issues like [DM-37970](https://jira.lsstcorp.org/browse/DM-37970) and [DM-37970](https://jira.lsstcorp.org/browse/DM-37970).